prov/efa/src/rdm/efa_rdm_ope.c

```diff
@@ -1909,3 +1909,48 @@ ssize_t efa_rdm_ope_repost_ope_queued_before_handshake(struct efa_rdm_ope *ope)
 		return -FI_EINVAL;
 	}
 }
+
+inline int efa_rdm_ope_process_queued_ope(struct efa_rdm_ope *ope, uint16_t flag)
+{
+	int ret = 0;
+
+	assert(flag & EFA_EDM_OPE_QUEUED_FLAGS);
+
+	if (!(ope->internal_flags & flag))
+		return -FI_EAGAIN;
+
+	switch (flag) {
+	case EFA_RDM_OPE_QUEUED_BEFORE_HANDSHAKE:
+		ret = efa_rdm_ope_repost_ope_queued_before_handshake(ope);
+		--ope->ep->ope_queued_before_handshake_cnt;
+		break;
+	case EFA_RDM_OPE_QUEUED_RNR:
+		assert(!dlist_empty(&ope->queued_pkts));
+		ret = efa_rdm_ep_post_queued_pkts(ope->ep, &ope->queued_pkts);
+		break;
+	case EFA_RDM_OPE_QUEUED_CTRL:
+		ret = efa_rdm_ope_post_send(ope, ope->queued_ctrl_type);
+		break;
+	case EFA_RDM_OPE_QUEUED_READ:
+		ret = efa_rdm_ope_post_read(ope);
+		break;
+	default:
+		break;
+	}
+
+	if (OFI_UNLIKELY(ret)) {
+		if (ret == -FI_EAGAIN)
+			return ret;
+
+		assert(ope->type == EFA_RDM_TXE || ope->type == EFA_RDM_RXE);
+		if (ope->type == EFA_RDM_TXE)
+			efa_rdm_txe_handle_error(ope, -ret, flag == EFA_RDM_OPE_QUEUED_RNR ? FI_EFA_ERR_PKT_SEND : FI_EFA_ERR_PKT_POST);
```

Comment:
I am not sure we need to have these differentiation. IMO `FI_EFA_ERR_PKT_POST` should cover `FI_EFA_ERR_PKT_SEND` and `FI_EFA_ERR_READ_POST`. The old code even has 3 error code, `FI_EFA_ERR_PKT_SEND`, `FI_EFA_ERR_PKT_POST`, `FI_EFA_ERR_READ_POST`. 

I cannot tell the difference between `FI_EFA_ERR_PKT_POST` and `FI_EFA_ERR_PKT_SEND` at least.

For simplicity I would just use `FI_EFA_ERR_PKT_POST` for all post failure in this function